clippy_lints/src/unwrap.rs

```diff
@@ -192,6 +195,43 @@ fn collect_unwrap_info<'tcx>(
     Vec::new()
 }
 
+/// A HIR visitor delegate that checks if a local variable of type `Option<_>` is mutated,
+/// unless `Option::as_mut` is called.
+struct MutationVisitor<'tcx> {
+    is_mutated: bool,
+    local_id: HirId,
+    tcx: TyCtxt<'tcx>,
+}
+
+fn is_option_as_mut_use(tcx: TyCtxt<'_>, expr_id: HirId) -> bool {
```

Comment:
Could you document this function, even a single line will make it easier to understand.

Reply:
Added some more documentation, that should hopefully clarify it a bit. Is is this better?

Reply:
Yep, it's better ❤️